pennylane/qchem/openfermion_obs.py

```diff
@@ -1077,11 +1068,16 @@ def _molecular_hamiltonian(
             raise ValueError(
                 "Only 'jordan_wigner' mapping is supported for the differentiable workflow."
             )
-        if mult != 1:
+        nuclear_charges = [atomic_numbers[s] for s in symbols]
+
+        n_electrons = sum(nuclear_charges) - charge
```

Comment:
```suggestion
        n_electrons = sum([atomic_numbers[s] for s in symbols]) - charge
```

Reply:
I think this would work even with the generator - ```sum(atomic_numbers[s] for s in symbols)```

Reply:
@obliviateandsurrender I don't think I follow your comment.